src/BaseMessage.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Yiisoft\Mailer;
+
+/**
+ * BaseMessage serves as a base class that implements the [[send()]] method required by [[MessageInterface]].
+ *
+ * By default, [[send()]] will use the "mail" application component to send the current message.
+ * The "mail" application component should be a mailer instance implementing [[MailerInterface]].
+ *
+ * @see BaseMailer
+ */
+abstract class BaseMessage implements MessageInterface
+{
+    /**
+     * @var MailerInterface the mailer instance that created this message.
+     * For independently created messages this is `null`.
+     */
+    private $mailer;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setMailer(MailerInterface $mailer)
+    {
+        $this->mailer = $mailer;
+    }
+
+    /**
+     * Sends this email message.
+     * @param MailerInterface $mailer the mailer that should be used to send this message.
+     * If no mailer is given it will first check if [[mailer]] is set and if not,
+     * the "mail" application component will be used instead.
+     * @return bool whether this message is sent successfully.
+     */
+    public function send(MailerInterface $mailer = null): bool
```

Comment:
Do we need this shortcut?

Reply:
Yes, it was used in method chaining: `$mailer->compose()->...->send()`, but the optional mailer parameter does not make sense for me, should we remove it?

Reply:
Yes.